silq/analysis/fit_toolbox.py

```diff
@@ -21,16 +21,45 @@ class Fit():
     This will find its initial parameters via `Fit.find_initial_parameters`,
     after which it will fit the data to `Fit.fit_function`.
 
+    If either xvals and ydata are passed, or only ydata is passed but is is a
+    Qcodes DataArray, the fit is automatically performed.
+    Otherwise Fit.perform_fit must be called.
+
+    Args:
+        ydata: Data values to be fitted
+        xvals: Sweep values, automatically extracted from ydata if not
+        explicitly provided and ydata is a DataArray.
+        fit: Automatically perform a fit if ydata is passed
```

Comment:
Rarely, only times I can think of is if I want to perform some customization on the initial parameters such as adding bounds. It can now all be passed as kwargs, but it might be more convenient to do it separately. I'm undecided about this one